modules/users/client/controllers/profile-edit-account.client.controller.js

```diff
@@ -41,6 +43,41 @@
 
     activate();
 
+    /**
+     * Parse $error and return a string
+     * @param {Object} usernameModel - Angular model for username form input
+     * @returns {String} error text
+     */
+    function getUsernameValidationError(usernameModel) {
+      if (!usernameModel || !usernameModel.$dirty || usernameModel.$valid) {
+        return '';
+      }
+
+      var err = usernameModel.$error || {};
+
+      if (err.required || usernameModel.$usernameValue === '') {
+        return 'Username is required.';
+      }
+
+      if (err.maxlength) {
+        return 'Too long, maximum length is 34 characters.';
+      }
+
+      if (err.minlength) {
+        return 'Too short, minumum length is 3 characters.';
+      }
+
+      if (err.pattern) {
+        return 'Invalid username.';
+      }
+
+      if (err.username) {
+        return 'This username is already in use or invalid.';
+      }
+
+      return 'Invalid username.';
+    }
```

Comment:
I've basically duplicated the code in the [signup controller](https://github.com/Trustroots/trustroots/blob/3825ed65472d9e6290d108c84b13cdb60d7eac1e/modules/users/client/controllers/signup.client.controller.js#L62-L96). In an ideal world, the whole input would just be a directive that we could re-use both here and at signup:

```html
<tr-username-input ng-model="username">
```

...to keep things simple, that should be a separate PR.